### DISCUSSION TITLE
How to skip the `<body>` element from generated HTML

### DISCUSSION THREAD

**Discussion by padcom** [GrapesJS/grapesjs#5418] [Q&A] [answered]
I am using the `editor.toHtml()` without JavaScript (`jsInHtml: false` config option). When calling the `editor.getHtml()` method the generated HTML contains the wrapping element, which by default is `<body>`. I was wondering if there is a way for the generated HTML to not include the `<body>` wrapper, so that I can have my own wrapper, or even have the generated HTML be part of some structure.

I did find a way to unwrap it in the browser by creating an element, assigning its `innerHTML`, getting the `querySelect('body')` result and using that's body's `innerHTML`. But that's like putting my right hand in the left pocket and I feel there should be a simpler way to do it.

**Comment by artf** [comment, marked as answer]
The body is the `wrapper` component, so you can extend it as you wish, for example:
```js
editor.Components.addType('wrapper', {
 model: {
   defaults: {
     tagName: 'div', // use `div` instead of `body` in the HTML output
   },

   // Skip wrapper in the HTML output
   toHTML(opts) {
     return this.getInnerHTML(opts);
   }
 }
})
```

**Reply by padcom** [reply]
That thing with `toHTML(...)` is _exactly_ what I needed. Thanks!
